deploy.py

```diff
@@ -284,84 +513,219 @@ def verify_flashed_app(self, expected_app):
+          if device.vendor_id == "1915" and device.product_id == "521F":
+            serial_number.append(device.serial_number)
+        if not serial_number:
+          fatal("Couldn't find any DFU device on your system.")
+        if len(serial_number) > 1:
+          fatal("Multiple DFU devices are detected. Please only connect one.")
+        # Run the command without capturing stdout so that we show progress
+        info("Flashing device using DFU...")
+        return subprocess.run(
+            [
+                "nrfutil", "dfu", "usb-serial",
+                "--package={}".format(dfu_pkg_file),
+                "--serial-number={}".format(serial_number[0])
+            ],
+            check=False,
+            timeout=None,
+        ).returncode
     return 0
 
 
 def main(args):
   # Make sure the current working directory is the right one before running
   os.chdir(os.path.realpath(os.path.dirname(__file__)))
-  # Check for pre-requisite executable files.
-  if not shutil.which("JLinkExe"):
-    fatal(("Couldn't find JLinkExe binary. Make sure Segger JLink tools "
-           "are installed and correctly set up."))
 
   OpenSKInstaller(args).run()
 
 
 if __name__ == "__main__":
-  shared_parser = argparse.ArgumentParser(add_help=False)
-  shared_parser.add_argument(
+  main_parser = argparse.ArgumentParser()
+  action_group = main_parser.add_mutually_exclusive_group(required=True)
+  action_group.add_argument(
+      "--list",
+      metavar="WHAT",
+      choices=("boards", "programmers"),
+      default=None,
+      dest="listing",
+      help=("List supported boards or programmers, 1 per line and then exit."),
+  )
+  action_group.add_argument(
+      "--board",
+      metavar="BOARD_NAME",
+      dest="board",
+      default=None,
+      choices=get_supported_boards(),
+      help="Indicates which board Tock OS will be compiled for.",
+  )
+
+  main_parser.add_argument(
       "--dont-clear-apps",
       action="store_false",
       default=True,
       dest="clear_apps",
       help=("When installing an application, previously installed "
             "applications won't be erased from the board."),
   )
+  main_parser.add_argument(
+      "--programmer",
+      metavar="METHOD",
+      dest="programmer",
+      choices=PROGRAMMERS,
+      default="jlink",
+      help=("Sets the method to be used to flash Tock OS or the application "
+            "on the target board."),
+  )
 
-  main_parser = argparse.ArgumentParser()
-  commands = main_parser.add_subparsers(
-      dest="action",
-      help=("Indicates which part of the firmware should be compiled and "
-            "flashed to the connected board."))
-
-  os_commands = commands.add_parser(
-      "os",
-      parents=[shared_parser],
-      help=("Compiles and installs Tock OS. The target board must be "
-            "specified by setting the --board argument."),
+  main_parser.add_argument(
+      "--no-tockos",
```

Comment:
This partially replies to my previous comment. Is there a `--no-app` option as well?

Reply:
I can add it.

Reply:
This saves time when fixing bugs in the kernel. On the other hand, calling `make flash` directly in the relevant `board/` folder of Tock allows to do the same. So it's not of high priority.

Reply:
Added the `--no-app` flag.